### PR TITLE
Use show table extended with table name list for get_catalog.

### PR DESCRIPTION
**Pull request by ueshin**
### Description

Uses `show table extended` with table name list for `get_catalog`.

- Running `describe table extended` for all tables could be slower than `show table extended` with table name list.
- Statistics that will appear in the generated docs are not included in `describe table extended`.

### CODE REVIEW

File: dbt/adapters/databricks/impl.py [hunk 1]
```diff
@@ -139,6 +146,47 @@ def list_relations_without_caching(  # type: ignore[override]
...
+    ) -> List[Tuple[DatabricksRelation, str]]:
+        kwargs = {"schema_relation": schema_relation}
+        try:
+            # The catalog for `show table extended` needs to match the current catalog.
+            with self._catalog(schema_relation.database):
+                results = self.execute_macro(SHOW_TABLE_EXTENDED_MACRO_NAME, kwargs=kwargs)
+        except dbt.exceptions.RuntimeException as e:
+            errmsg = getattr(e, "msg", "")
+            if (
+                "[SCHEMA_NOT_FOUND]" in errmsg
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/impl.py:160]: If there is an error code in the error message, will we get the error message details as well ("Database ... not found")?

File: dbt/adapters/databricks/column.py [hunk 2]
```diff
@@ -1,9 +1,22 @@
 from dataclasses import dataclass
+from typing import ClassVar, Dict
 
 from dbt.adapters.spark.column import SparkColumn
 
 
 @dataclass
 class DatabricksColumn(SparkColumn):
+    TYPE_LABELS: ClassVar[Dict[str, str]] = {
+        "LONG": "BIGINT",
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/column.py:10]: Why do we need this?

**Review by allisonwang-db — COMMENTED** [review]
Looks good!

File: dbt/adapters/databricks/impl.py (hunk 1 above)
Comment (ueshin) [reply dbt/adapters/databricks/impl.py:160]: The error message was changed:

> [SCHEMA_NOT_FOUND] The schema \<schemaName> cannot be found. Verify the spelling and correctness of the schema and catalog.
If you did not qualify the name with a catalog, verify the current_schema() output, or qualify the name with the correct catalog.
To tolerate the error on drop use DROP SCHEMA IF EXISTS.

File: dbt/adapters/databricks/column.py (hunk 2 above)
Comment (ueshin) [reply dbt/adapters/databricks/column.py:10]: This is a port of https://github.com/dbt-labs/dbt-spark/pull/358.

**Review by allisonwang-db — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**